File: 167/m167.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution
{
public:
	vector<int> twoSum(vector<int>& numbers, int target)
	{
		vector<int>::iterator it1 = numbers.begin(), it2 = numbers.end() - 1;
		int idx1 = 1, idx2 = numbers.size(), nSum;

		while (it1 != it2)
		{
			nSum = *it1 + *it2;
			if (nSum == target) return { idx1, idx2 };
			else if (nSum < target) it1++, idx1++;
			else it2--, idx2--;
		}

		return { 0, 0 };
	}
};

int main()
{
	Solution x;
	vector<int> stockPrices = { 2,7,11,15 };
	//vector<string> res;

	auto res = x.twoSum(stockPrices, 13);
}

File: 01/67/m167.swift
// Solved

class Solution {
	func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
		var l = 0, r = numbers.count - 1
		
		while l < r {
			let s = numbers[l] + numbers[r]
			if s == target {
				return [l + 1, r + 1]
			}
			if s < target {
				l += 1
			}
			else {
				r -= 1
			}
		}
		
		return [-1]
	}
}
